### ISSUE TITLE
MAE Consumer fails to start with Analytics enabled

### ISSUE THREAD

**Issue by marcossantiago** [linkedin/datahub#3097]
**Describe the bug**
I have just upgraded from `v0.8.6` to `v0.8.9` but the MAE Consumer pods keep crashing with error below. It works great in the previous version and works fine with `datahub_analytics_enabled: false`. 

Can someone point to the right direction on how to fix it ?

Happy to provide more information if needed. 

**To Reproduce**
1. `v0.8.9` k8s deployment using latest [helm-chart](https://github.com/acryldata/datahub-helm/tree/datahub-0.2.13/charts/datahub)
2. Enable Analytics using `datahub_analytics_enabled: false`

**Expected behavior**
MAE Consumer pod works properly

**Additional context**
Error log:

```
2021/08/13 16:33:59 Waiting for: tcp://kafka-xxxxxxxx:9094
2021/08/13 16:33:59 Waiting for: https://elasticsearchxxxxxx:443
2021/08/13 16:33:59 Connected to tcp://kafka-xxxxxxxx:9094
2021/08/13 16:33:59 Received 200 from https://elasticsearchxxxxxx:443

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::        (v2.1.4.RELEASE)

16:34:00.510 [main] INFO  c.l.m.kafka.MaeConsumerApplication - Starting MaeConsumerApplication on datahub-datahub-mae-consumer-7c58479b86-d496n with PID 37 (/datahub/datahub-mae-consumer/bin/mae-consumer-job.jar started by datahub in /)
16:34:00.515 [main] INFO  c.l.m.kafka.MaeConsumerApplication - No active profile set, falling back to default profiles: default
16:34:00.890 [background-preinit] WARN  o.s.h.c.j.Jackson2ObjectMapperBuilder - For Jackson Kotlin classes support please add "com.fasterxml.jackson.module:jackson-module-kotlin" to the classpath
16:34:01.836 [main] INFO  o.s.c.s.PostProcessorRegistrationDelegate$BeanPostProcessorChecker - Bean 'org.springframework.kafka.annotation.KafkaBootstrapConfiguration' of type [org.springframework.kafka.annotation.KafkaBootstrapConfiguration$$EnhancerBySpringCGLIB$$89b87979] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
16:34:02.067 [main] INFO  org.eclipse.jetty.util.log - Logging initialized @2798ms to org.eclipse.jetty.util.log.Slf4jLog
16:34:02.199 [main] INFO  o.s.b.w.e.j.JettyServletWebServerFactory - Server initialized with port: 9091
16:34:02.203 [main] INFO  org.eclipse.jetty.server.Server - jetty-9.4.15.v20190215; built: 2019-02-15T16:53:49.381Z; git: eb70b240169fcf1abbd86af36482d1c49826fa0b; jvm 1.8.0_292-b10
16:34:02.270 [main] INFO  o.e.j.s.h.ContextHandler.application - Initializing Spring embedded WebApplicationContext
16:34:02.270 [main] INFO  o.s.web.context.ContextLoader - Root WebApplicationContext: initialization completed in 1689 ms
16:34:02.864 [main] INFO  org.eclipse.jetty.server.session - DefaultSessionIdManager workerName=node0
16:34:02.864 [main] INFO  org.eclipse.jetty.server.session - No SessionScavenger set, using defaults
16:34:02.866 [main] INFO  org.eclipse.jetty.server.session - node0 Scavenging every 600000ms
16:34:02.877 [main] INFO  o.e.j.server.handler.ContextHandler - Started o.s.b.w.e.j.JettyEmbeddedWebAppContext@5f049ea1{application,/,[file:///tmp/jetty-docbase.2756965692778223956.9091/],AVAILABLE}
16:34:02.877 [main] INFO  org.eclipse.jetty.server.Server - Started @3611ms
16:34:03.006 [main] WARN  c.l.r.t.h.client.HttpClientFactory - No scheduled executor is provided to HttpClientFactory, using it's own scheduled executor.
16:34:03.006 [main] WARN  c.l.r.t.h.client.HttpClientFactory - No callback executor is provided to HttpClientFactory, using it's own call back executor.
16:34:03.006 [main] WARN  c.l.r.t.h.client.HttpClientFactory - No Compression executor is provided to HttpClientFactory, using it's own compression executor.
16:34:03.010 [main] INFO  c.l.r.t.h.client.HttpClientFactory - The service 'null' has been assigned to the ChannelPoolManager with key 'noSpecifiedNamePrefix 1138266797 '
16:34:04.141 [main] WARN  o.s.b.w.s.c.AnnotationConfigServletWebServerApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dataHubUsageEventsProcessor' defined in URL [jar:file:/datahub/datahub-mae-consumer/bin/mae-consumer-job.jar!/BOOT-INF/lib/mae-consumer.jar!/com/linkedin/metadata/kafka/DataHubUsageEventsProcessor.class]: Initialization of bean failed; nested exception is org.springframework.beans.factory.BeanInitializationException: Could not register Kafka listener endpoint on [public void com.linkedin.metadata.kafka.DataHubUsageEventsProcessor.consume(org.apache.kafka.clients.consumer.ConsumerRecord)] for bean dataHubUsageEventsProcessor, no KafkaListenerContainerFactory with id 'simpleKafkaConsumer' was found in the application context; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean named 'simpleKafkaConsumer' available
16:34:04.148 [main] INFO  org.eclipse.jetty.server.session - node0 Stopped scavenging
16:34:04.150 [main] INFO  o.e.j.server.handler.ContextHandler - Stopped o.s.b.w.e.j.JettyEmbeddedWebAppContext@5f049ea1{application,/,[file:///tmp/jetty-docbase.2756965692778223956.9091/],UNAVAILABLE}
16:34:04.163 [main] INFO  o.s.b.a.l.ConditionEvaluationReportLoggingListener -

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
16:34:04.242 [main] ERROR o.s.b.d.LoggingFailureAnalysisReporter -

***************************
APPLICATION FAILED TO START
***************************

Description:

A component required a bean named 'simpleKafkaConsumer' that could not be found.


Action:

Consider defining a bean named 'simpleKafkaConsumer' in your configuration.

2021/08/13 16:34:04 Command exited with error: exit status 1
```

**Comment by marcossantiago** [comment]
The mae-consumer manifest:

```
# Source: datahub/charts/datahub-mae-consumer/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: datahub-datahub-mae-consumer
  namespace: datahighway-dev
  labels:
    helm.sh/chart: datahub-mae-consumer-0.2.1
    app.kubernetes.io/name: datahub-mae-consumer
    app.kubernetes.io/instance: datahub
    app.kubernetes.io/version: "0.3.1"
    app.kubernetes.io/managed-by: Helm
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: datahub-mae-consumer
      app.kubernetes.io/instance: datahub
  template:
    metadata:
      labels:
        app.kubernetes.io/name: datahub-mae-consumer
        app.kubernetes.io/instance: datahub
    spec:
      hostAliases:
        - hostnames:
          - broker
          - mysql
          - elasticsearch
          - neo4j
          ip: 192.168.0.104
      serviceAccountName: datahub-datahub-mae-consumer
      securityContext:
        {}
      volumes:
      initContainers:
      containers:
        - name: datahub-mae-consumer
          securityContext:
            {}
          image: "linkedin/datahub-mae-consumer:v0.8.9"
          imagePullPolicy: Always
          ports:
            - name: http
              containerPort: 9091
              protocol: TCP
          livenessProbe:
            httpGet:
              path: /actuator/health
              port: http
            initialDelaySeconds: 60
            periodSeconds: 30
            failureThreshold: 8
          readinessProbe:
            httpGet:
              path: /actuator/health
              port: http
            initialDelaySeconds: 60
            periodSeconds: 30
            failureThreshold: 8
          env:
            - name: MAE_CONSUMER_ENABLED
              value: "true"
            - name: ENTITY_REGISTRY_CONFIG_PATH
              value: /datahub/datahub-mae-consumer/resources/entity-registry.yml
            - name: GMS_HOST
              value: datahub-datahub-gms
            - name: GMS_PORT
              value: "8080"
            - name: KAFKA_BOOTSTRAP_SERVER
              value: "kafkaxxxxxx:9094"
            - name: KAFKA_SCHEMAREGISTRY_URL
              value: "schema-regitryxxxxx"
            - name: ELASTICSEARCH_HOST
              value: "elasticsearchxxxxxx"
            - name: ELASTICSEARCH_PORT
              value: "443"
            - name: ELASTICSEARCH_USE_SSL
              value: "true"
            - name: GRAPH_SERVICE_IMPL
              value: elasticsearch
            - name: DATAHUB_ANALYTICS_ENABLED
              value: "true"
            - name: SPRING_KAFKA_PROPERTIES_SASL_JAAS_CONFIG
              value: org.apache.kafka.common.security.scram.ScramLoginModule required serviceName="datahub" username="userxxxxx" password="xxxxxxx";
            - name: SPRING_KAFKA_PROPERTIES_SASL_MECHANISM
              value: SCRAM-SHA-256
            - name: SPRING_KAFKA_PROPERTIES_SECURITY_PROTOCOL
              value: SASL_SSL
          volumeMounts:
          resources:
            {}
```

**Comment by marcossantiago** [comment]
Thanks @shirshanka! This issue has been fixed in the version `v0.8.10`. 🥳
